pkg/trantor/types/types.go

```diff
@@ -9,10 +10,41 @@ import (
 // ================================================================================
 
 // VoteWeight represents the weight of a node's vote when gathering quorums.
-type VoteWeight uint64
+// The underlying type is a string containing a decimal integer representation of the weight.
+// This is required to support large integers that would not fit in the native uint64 data type
+// which can occur if cryptocurrency units are used as weights in a PoS-like system.
+// When performing mathematical operations on weights, we convert them to the big.Int type.
+// We do not store them as big.Int, since that would make it harder to use them in protocol buffers.
+type VoteWeight string
 
 func (vw VoteWeight) Bytes() []byte {
-	return serializing.Uint64ToBytes(uint64(vw))
+	return []byte(vw)
+}
+
+func (vw VoteWeight) Pb() string {
+	return string(vw)
+}
+
+func (vw VoteWeight) String() string {
+	return string(vw)
+}
+
+func (vw VoteWeight) IsValid() bool {
+	if _, ok := new(big.Int).SetString(vw.String(), 10); !ok {
+		return false
+	}
+	return true
+}
+
+// BigInt converts a VoteWeight (normally represented as a string) to a big.Int.
+// BigInt panics if the underlying string is not a valid decimal representation of an integer.
+// Thus, BigInt must not be called on received input without having validated VoteWeight by calling the IsValid method.
+func (vw VoteWeight) BigInt() *big.Int {
```

Comment:
What about having a `BigInt()` that returns an error and doesn't panic, and then a `MustParseBigInt()` that panics? I am usually wary of assuming that a user (like me) will read the comments and use the function properly. I would assume that it can't error if I don't see an error in the signature.

(Weak opinion weakly held, feel free to disregard)